packit/local_project.py

```diff
@@ -578,8 +579,39 @@ def commit(
             allow_empty=allow_empty, m=message, amend=amend, **other_message_kwargs
         )
 
-    def get_commits(self, ref: str = "HEAD") -> Iterator[git.Commit]:
-        return self.git_repo.iter_commits(ref)
+    def get_commits(self, ref: str = "HEAD", **kwargs) -> Iterator[git.Commit]:
+        return self.git_repo.iter_commits(ref, **kwargs)
+
+    @staticmethod
+    def get_commit_diff(commit: git.Commit) -> List[git.Diff]:
+        """Get modified files of the given commit."""
+        if len(commit.parents) == 1:
+            return commit.parents[0].diff(commit, create_patch=True)
+        elif len(commit.parents) == 0:
+            # First commit in the repo
+            return commit.diff(git.NULL_TREE, create_patch=True)
+        else:
+            # Probably a merge commit, we can't do much about it
+            return []
+
+    def get_commit_hunks(self, commit: git.Commit) -> List[str]:
+        """Get a list of hunks of the given commit."""
+        patch = self.git_repo.git.show(commit, format="", color="never")
+        hunk_start_re = re.compile(r"diff --git a/[\w/]* b/[\w/]*")
```

Comment:
I think this regex will miss paths which have a `.` in them, like `tests/tests.yaml`. Updating one of the `get_commit_hunks` tests will reflect this, and something like this will resolve it:

```suggestion
        hunk_start_re = re.compile(r"diff --git a/.+ b/.+")
```

Due to this issue I see an `list index out of range` error when trying to convert a dist-git commit which modified `.gitignore` and `tests/tests.yaml` in the same `acl` repo I already mentioned.

Reply:
Good catch, fixed, thanks.